networking/firmware_protocol/src/lib.rs

```diff
@@ -90,17 +77,51 @@ impl SlimeString {
 }
 
 #[derive(Debug, PartialEq, DekuRead, DekuWrite)]
-#[deku(type = "u32")]
 #[deku(endian = "big")]
+pub struct Packet {
+	// TODO: This tag could really be dropped from the Rust side, but #[deku(temp)] is a bit wonky
+	tag: u32,
+	seq: u64,
+	#[deku(ctx = "*tag")]
+	ty_: PacketType,
+}
+
+impl Packet {
+	pub fn new(seq: u64, ty_: PacketType) -> Packet {
+		Packet {
+			tag: ty_.deku_id().unwrap(),
+			seq,
+			ty_,
+		}
+	}
+
+	pub fn serialize_into(&self, buf: &mut [u8]) -> usize {
```

Comment:
The current behaviour is a limitation of deku, since deku cannot serialize in-place. Ideally later we patch deku to support this or switch away from deku entirely. I felt it made more sense to hide the limitation as an implementation detail from the caller.